src/cljdoc/util/sqlite_cache.clj

```diff
@@ -81,7 +84,8 @@
   (lookup [_ k]
     (delay (fetch-item! k (:cache-spec state))))
   (lookup [this k not-found]
-    (delay (or (deref (cache/lookup this k))) not-found))
+    (delay (or (fetch-item! k (:cache-spec state))
+               (if (derefable? not-found) not-found (delay not-found)))))
```

Comment:
I'm still a little bit confused by this I think.

- It seems as if we will end up with a value that's wrapped in two `delay`s 
- Did you test these things in a REPL / what's your confidence this is correct?

Reply:
Oh gee, you know what? It does look odd. I think I made the correction to your review comment too hastily,  will take more careful look at this.

Reply:
Ok, I think this area deserved some unit tests so added some.
This led me to also find that `memoize/memo-clear!` was not working for clear all case so fixed that too. 
Changes are in this commit: 9c92e1ad0578720e7358ee21e0ec2306d1ba2bec